client/FxAccountClient.js

```diff
@@ -1031,4 +1099,28 @@ define([
   return FxAccountClient;
 });
 
+var METRICS_CONTEXT_FIELDS = {
+  context: true,
+  entrypoint: true,
+  flowId: true,
+  migration: true,
+  service: true,
+  utmCampaign: true,
+  utmContent: true,
+  utmMedium: true,
+  utmSource: true,
+  utmTerm: true
+};
+
+function marshallMetricsContext (data) {
+  var metricsContext = {};
+
+  Object.keys(data).forEach(function (key) {
```

Comment:
If we are planning to get things working in es5 in here: https://github.com/mozilla/fxa-content-server/issues/2886 it might be worth keeping this <es5. Need to discuss...